service/matching/matching_engine.go

```diff
@@ -950,8 +950,24 @@ func (e *matchingEngineImpl) DescribeTaskQueue(
 					if physInfo, ok := physicalInfoByBuildId[buildId][taskQueueType]; !ok {
 						physicalInfoByBuildId[buildId][taskQueueType] = vii.PhysicalTaskQueueInfo
 					} else {
+						var mergedStats *taskqueuepb.TaskQueueStats
+
+						// only report Task Queue Statistics if requested.
+						if req.GetReportStats() {
+							totalStats := physicalInfoByBuildId[buildId][taskQueueType].TaskQueueStats
+							partitionStats := vii.PhysicalTaskQueueInfo.TaskQueueStats
+
+							// Aggregate counts; for now, we only aggregate approximateBacklogCount
+							mergedStats = &taskqueuepb.TaskQueueStats{
+								ApproximateBacklogCount: totalStats.ApproximateBacklogCount + partitionStats.ApproximateBacklogCount,
+								ApproximateBacklogAge:   nil,
+								TasksAddRate:            float32(0),
+								TasksDispatchRate:       float32(0),
+							}
+						}
 						merged := &taskqueuespb.PhysicalTaskQueueInfo{
```

Comment:
I think I was reading the wrong before.. why do we create a new taskqueuespb.PhysicalTaskQueueInfo each time and just set it back in the map? it seems like this is doing more allocation than necessary.

not to fix in this PR, just keep in mind for late. @carlydf